test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -680,6 +681,28 @@ public function testIsAnonymousWithAnonymousClassNoNamespace() : void
         self::assertStringEndsWith('Fixture/AnonymousClassNoNamespace.php(3)', $classInfo->getName());
     }
 
+    public function testIsAnonymousWithParentClass() : void
+    {
+        $reflector = new ClassReflector(
+            new SingleFileSourceLocator(
+                realpath(__DIR__ . '/../Fixture/ClassForHinting.php'),
+                $this->astLocator
+            )
+        );
+
+        $anonymousClassInfo = ReflectionClass::createFromInstance(
+            new class extends ClassForHinting {
+                public function a() : void
+                {
+                }
+            }
+        );
+
+        $parent = $anonymousClassInfo->getParentClass();
+        self::assertSame(ClassForHinting::class, $parent->getName());
+        self::assertSame((string) $parent, (string) $reflector->reflect(ClassForHinting::class));
```

Comment:
@asgrim can you check this as well? I'm unsure about the test behavior, think this should be a `self::assertEquals()?